core/iwasm/libraries/lib-socket/test/tcp.c

```diff
@@ -0,0 +1,90 @@
+#include <unistd.h>
+#include <string.h>
+#include "assert.h"
+#ifdef __wasi__
+#include <wasi_socket_ext.h>
+#else
+#include <arpa/inet.h>
+#endif
+#include <pthread.h>
+
+#define PORT 8989
+#define IP "127.0.0.1"
+
+struct init_socket {
+    struct sockaddr_in addr;
+    int sock;
+};
+
+struct init_socket
+init_socket_addr()
+{
+    int sock = socket(AF_INET, SOCK_STREAM, 0);
+    assert(sock > 0);
+
+    struct sockaddr_in addr;
+    memset(&addr, '\0', sizeof(addr));
+    addr.sin_family = AF_INET;
+    addr.sin_port = PORT;
+    addr.sin_addr.s_addr = inet_addr(IP);
```

Comment:
Consider using `htonl(INADDR_LOOPBACK);` instead, so we don't have to hardcode IP address.